src/admin-dhis2/components/AdminDhis2PeriodMappingForm/AdminDhis2PeriodMappingForm.jsx

```diff
@@ -26,48 +28,63 @@ function AdminDhis2PeriodMappingForm({ onSubmit, onCancel, serverId }) {
     const setInitialValues = () => {
         setProvidedName('');
         setSelectedSource('');
-        setSelectedDhisPeriodType('');
+        setSelectedProcessingPeriod(undefined);
         setSelectedProcessingSchedule(undefined);
         setErrors([]);
     };
 
-    const fetchProcessingPeriods = () => {
-        serverService.getProcessingSchedules()
-            .then((fetchedProcessingPeriods) => {
-                const { content } = fetchedProcessingPeriods;
+    const fetchServerDataset = () => {
+        serverService.getServerDataset(serverId, datasetId)
+            .then((fetchedServerDataset) => {
+                setDhisPeriod(fetchedServerDataset.periodType);
+            })
+    }
 
+    const fetchProcessingSchedules = () => {
+        serverService.getProcessingSchedules()
+            .then((fetchedProcessingSchedules) => {
+                const { content } = fetchedProcessingSchedules;
+                
                 const elements = content.map((element) => ({
                     name: element.name,
-                    value: element.id,
-                    startDate: element.startDate,
-                    endDate: element.endDate
+                    value: element.id
                 }));
+
                 setProcessingSchedulesOptions(elements);
             });
     }
 
-    const fetchDhisPeriodTypes = () => {
-        if (serverId) {
-            serverService?.getDhisPeriodTypes(serverId)
-            .then((fetchedDhisPeriodTypes) => {
-
-                const dhisPeriodTypes = fetchedDhisPeriodTypes.map((dhisPeriodType) => ({
-                    name: dhisPeriodType.name,
-                    value: dhisPeriodType.name,
-                }));
-                setDhisPeriodTypesOptions(dhisPeriodTypes);
+    const fetchProcessingPeriods = () => {
+        serverService.getProcessingPeriods()
+            .then((fetchedProcessingPeriods) => {
+                const { content } = fetchedProcessingPeriods;
+                setProcessingPeriods(content);
             });
-        }
+    }
+
+    const filterProcessingPeriodsBySchedule = (scheduleId) => {
+        const processingPeriodsWithScheduleId = processingPeriods
+            .filter((processingPeriod) => processingPeriod.processingSchedule.id === scheduleId)
+            .map((processingPeriod) => ({
+                name: processingPeriod.name,
+                value: processingPeriod.id,
+                startDate: processingPeriod.startDate,
+                endDate: processingPeriod.endDate,
+                processingScheduleId: processingPeriod.processingSchedule.id
+            }))
+            .slice(0,4);
```

Comment:
please create constants and name it properly to be more transparent what are you doing here